public/js/helpers/chart_helper.js

```diff
@@ -53,3 +53,101 @@ export function padPoints (pts, binSize, sustain) {
   pts.unshift(front)
   pts.push(back)
 }
+
+var zoomMap = {
+  all: 0,
+  year: 3.154e+10,
+  month: 2.628e+9,
+  week: 6.048e+8,
+  day: 8.64e+7
+}
+
+export class Zoom {
+  static object (start, end) {
+    return {
+      start: start,
+      end: end
+    }
+  }
+
+  static map (key) {
+    return zoomMap[key]
+  }
+
+  static encode (start, end) {
+    if (!end) {
+      start = this.tryDecode(start)
+      end = start.end
+      start = start.start
+    }
+    return parseInt(start / 1000).toString(36) + '-' + parseInt(end / 1000).toString(36)
+  }
+
+  static decode (encoded, lims) {
+    // if lims are provided, encoded can be a zoomMap key
+    encoded = this.tryDecode(encoded)
```

Comment:
In general, I think it's easier to reason about a codebase where we pretend that the inputs to functions are immutable, and always return a new value. So instead of overwriting the input, do:
`let decoded = this.tryDecode(encoded)`